libs/hdf_converters/src/aws-config-mapper.ts

```diff
@@ -0,0 +1,345 @@
+  private hdfTags(configRule: ConfigRule): Record<string, unknown> {
+    let result = {};
+    const sourceIdentifier = configRule.Source?.SourceIdentifier;
+    result = _.set(result, 'nist', []);
+    let defaultMatch: string[] | null = [];
+    if (sourceIdentifier !== undefined) {
+      defaultMatch = AWS_CONFIG_MAPPING.nistFilter([sourceIdentifier]);
+    }
+    if (Array.isArray(defaultMatch) && defaultMatch.length !== 0) {
+      result = _.set(
+        result,
+        'nist',
+        _.get(result, 'nist').concat(defaultMatch)
+      );
+    }
+
+    //Deal with custom mapping later
+    // custom_mapping_match = @custom_mapping[source_identifier]
+
+    // result['nist'] += custom_mapping_match['NIST-ID'].split('|').map { | name | "#{name} (user provided)"} unless custom_mapping_match.nil ?
+
+    if (
+      Array.isArray(_.get(result, 'nist')) &&
+      _.get(result, 'nist').length === 0
+    ) {
+      result = _.set(result, 'nist', ['unmapped']);
+    }
+    return result;
+  }
+  private checkText(configRule: ConfigRule): string {
+    let params: any[] = [];
+    if (
+      configRule.InputParameters !== undefined &&
+      configRule.InputParameters !== '{}'
+    ) {
+      params = configRule.InputParameters.replace(/{/gi, '')
+        .replace(/}/gi, '')
+        .split(',');
+    }
+    const checkText = [];
+    checkText.push(`ARN: ${configRule.ConfigRuleArn || 'N/A'}`);
+    checkText.push(
+      `Source Identifier: ${configRule.Source?.SourceIdentifier || 'N/A'}`
+    );
+    if (params.length !== 0) {
+      checkText.push(`${params.join('<br/>')}`);
+    }
+    return checkText.join('<br/>');
+  }
+  private hdfDescriptions(configRule: ConfigRule) {
+    return [
+      {
+        data: this.checkText(configRule),
+        label: 'check'
+      }
+    ];
+  }
+  private getAccountId(arn: string): string {
+    const matches = arn.match(/:(\d{12}):config-rule/);
+    if (matches === null) {
+      return 'no-account-id';
+    } else {
+      return matches[0];
+    }
+  }
+  private async getControls(): Promise<ExecJSON.Control[]> {
+    return (await this.issues).map((issue) => {
+      return {
+        id: issue.ConfigRuleId || '',
+        title: `${this.getAccountId(issue.ConfigRuleArn || '')} - ${
+          issue.ConfigRuleName
+        }`,
+        desc: issue.Description,
+        impact: 0.5,
+        tags: this.hdfTags(issue),
+        descriptions: this.hdfDescriptions(issue),
+        refs: [],
+        source_location: {ref: issue.ConfigRuleArn, line: 1},
+        code: '',
+        results: _.get(issue, 'results')
+      };
+    });
+  }
+  public async toHdf(): Promise<ExecJSON.Execution> {
+    const hdf: ExecJSON.Execution = {
+      platform: {
+        name: 'Heimdall Tools',
+        release: HeimdallToolsVersion,
+        target_id: ''
+      },
+      version: HeimdallToolsVersion,
+      statistics: {
+        //aws_config_sdk_version: ConfigService., // How do i get the sdk version?
```

Comment:
Is this comment still relevant?

Reply:
Yes, since the 
ExecJSON type does not allow the extra field "aws_config_sdk_version", I need to find a workaround (will likely involve transforming/appending the field at the end after the result is generated)